easybuild/tools/containers.py

```diff
@@ -0,0 +1,305 @@
+        else:
+            raise EasyBuildError("Singularity base image at specified path does not exist: %s", base_image)
+
+    # otherwise, bootstrap agent is 'docker' or 'shub'
+    # format --container-base {docker|shub}:<image>:<tag>
+    else:
+        base_image = base_specs['arg1']
+        # image tag is optional
+        base_image_tag = base_specs.get('arg2', None)
+
+    bootstrap_from = base_image
+    if base_image_tag:
+        bootstrap_from += ':' + base_image_tag
+
+    # if there is osdependencies in easyconfig then add them to Singularity recipe
+    install_os_deps = ''
+    for ec in easyconfigs:
+        for osdep in ec['ec']['osdependencies']:
+            if isinstance(osdep, basestring):
+                install_os_deps += "yum install -y %s\n" % osdep
+            # tuple entry indicates multiple options
+            elif isinstance(osdep, tuple):
+                install_os_deps += "yum --skip-broken install %s\n" % ' '.join(osdep)
+            else:
+                raise EasyBuildError("Unknown format of OS dependency specification encountered: %s", osdep)
+
+    # module names to load in container environment
+    mod_names = [e['ec'].name + '/' + det_full_ec_version(e['ec']) for e in easyconfigs]
+
+    # name of Singularity definition file
+    img_name = build_option('container_image_name')
+    if img_name:
+        def_file_label = os.path.splitext(img_name)[0]
+    else:
+        def_file_label = mod_names[0].replace('/', '-')
+
+    def_file = 'Singularity.%s' % def_file_label
+
+    # adding all the regions for writing the  Singularity definition file
+    content = SINGULARITY_TEMPLATE % {
+        'bootstrap': bootstrap_agent,
+        'from': bootstrap_from,
+        'install_os_deps': install_os_deps,
+        'easyconfigs': ' '.join(os.path.basename(e['spec']) for e in easyconfigs),
+        'mod_names': ' '.join(mod_names),
+    }
+    def_path = os.path.join(cont_path, def_file)
+
+    if os.path.exists(def_path):
+        if build_option('force'):
+            print_msg("WARNING: overwriting existing container recipe at %s due to --force" % def_path)
+        else:
+            raise EasyBuildError("Container recipe at %s already exists, not overwriting it without --force", def_path)
+
+    write_file(def_path, content)
+    print_msg("Singularity definition file created at %s" % def_path, log=_log)
+
+    return def_path
+
+
+def build_singularity_image(def_path):
+    """Build Singularity container image by calling out to 'singularity' (requires admin privileges!)."""
+
+    cont_path = container_path()
+    def_file = os.path.basename(def_path)
+
+    # use --imagename if specified, otherwise derive based on filename of recipe
+    img_name = build_option('container_image_name')
+    if img_name is None:
+        # definition file Singularity.<app>-<version, container name <app>-<version>.<img|simg>
+        dot_idx = def_file.find('.')
+        img_name = def_file[dot_idx+1:]
```

Comment:
What about `img_name = def_file.split('.', 1)[1]`

Reply:
That's better indeed, will change.